Task_2_2_1

andrey-dru-me1 opened the pull request.

andrey-dru-me1 committed:
Try to resolve issues with JDK version

andrey-dru-me1 committed:
start implementing the task

andrey-dru-me1 committed:
implement all cook's work

andrey-dru-me1 committed:
implement courier's functionality

andrey-dru-me1 committed:
remove unused class

andrey-dru-me1 committed:
change build.gradle

andrey-dru-me1 committed:
change google_checks.xml

andrey-dru-me1 committed:
add json parser

andrey-dru-me1 committed:
add logback logger

andrey-dru-me1 committed:
add uml diagram

andrey-dru-me1 committed:
remove redundant comments

andrey-dru-me1 committed:
reformat Cook a little

molo4evan commented on Task_2_2_1/src/main/java/ru/nsu/fit/oop/melnikov/Logistics.java:
```diff
@@ -0,0 +1,82 @@
+
+  public Logistics(@NotNull Set<Cook> cooks, Set<Courier> couriers, WareHouse wareHouse) {
+    this.wareHouse = wareHouse;
+    this.orders = new int[1];
```
енто чего, енто зачем...

andrey-dru-me1 replied:
Вообще, мне это IDE'шка предложила, чтобы я мог синхронизироваться по `int`

andrey-dru-me1 replied:
Убрал к чертям Logistics класс, так что можно считать, что решено?

molo4evan commented on Task_2_2_1/src/main/java/ru/nsu/fit/oop/melnikov/WareHouse.java:
```diff
@@ -0,0 +1,31 @@
+
+import static java.lang.Math.min;
+
+public class WareHouse {
```
Warehouse

andrey-dru-me1 replied:
Исправил https://github.com/andrey-dru-me1/oop/pull/11/commits/8051a0d5095b43dd8d80997140872d0978062ba8

molo4evan commented on Task_2_2_1/src/main/java/ru/nsu/fit/oop/melnikov/WareHouse.java:
```diff
@@ -0,0 +1,31 @@
+  }
+
+  public synchronized int takePizzas(int amount) throws IndexOutOfBoundsException {
+    if (pizzasAmount <= 0) {
```
Ну зачем же так жестоко...
Мы же не пытаемся выгнать работников на мороз, они могут и в тепле подождать

andrey-dru-me1 replied:
Исправил

molo4evan commented on Task_2_2_1/src/main/java/ru/nsu/fit/oop/melnikov/Cook.java:
```diff
@@ -0,0 +1,48 @@
+  }
+
+  @SuppressWarnings("BusyWait")
+  public void work(Runnable iAmFree, Runnable stayAtWareHouse) {
```
не-не-не-не-не
У нас ООП
В ООП инкапсуляция
Пекарь сам знает, что ему делать, курьер сам знает, что ему делать, хранилище само знает, как ему работать, и "логистика" не пытается делать их работу за них

andrey-dru-me1 replied:
Переделал всю структуру проекта. Вроде сейчас такой жести нет)

molo4evan commented on Task_2_2_1/src/main/java/ru/nsu/fit/oop/melnikov/Logistics.java:
```diff
@@ -0,0 +1,82 @@
+import java.util.Set;
+import org.jetbrains.annotations.NotNull;
+
+public class Logistics {
```
Добавь Main с интерактивным режимом работы

andrey-dru-me1 replied:
Добавил

molo4evan commented on Task_2_2_1/src/test/java/ru/nsu/fit/oop/melnikov/logistics/LogisticsTest.java:
```diff
@@ -0,0 +1,59 @@
+import ru.nsu.fit.oop.melnikov.Logistics;
+import ru.nsu.fit.oop.melnikov.WareHouse;
+
+class LogisticsTest {
```
- Проверка корректной многопоточной работы с методами хранилища
- Проверка состояния пиццерии после soft shutdown, force shutdown

andrey-dru-me1 committed:
redo the whole logic and structure

andrey-dru-me1 committed:
implement simple interaction without exit

andrey-dru-me1 committed:
implement interruption

andrey-dru-me1 committed:
restructure code

andrey-dru-me1 committed:
Fix small issue

andrey-dru-me1 committed:
implement json loader

andrey-dru-me1 committed:
add new functionality: couriers waits for trunk filling now but some time later go fulfilling an order.

andrey-dru-me1 committed:
make json file loading easier

andrey-dru-me1 committed:
fix some warnings;
refactor customer

andrey-dru-me1 committed:
solve linter's warning

andrey-dru-me1 committed:
add main class tests

andrey-dru-me1 commented:
![image](https://user-images.githubusercontent.com/91488312/227257717-2ef2216a-4b5f-4f8b-84a7-9c9e5a6959e0.png)

andrey-dru-me1 committed:
add javadocs;
update gradle dependencies;
update gradle itself;
cancel division between cooks and couriers in ThreadController.

andrey-dru-me1 committed:
downgrade gradle version so test for task_2_1_1 could work

andrey-dru-me1 committed:
downgrade dependencies versions to make things work

andrey-dru-me1 committed:
change <li> to -

andrey-dru-me1 committed:
implement soft and hard interrupts

andrey-dru-me1 committed:
improve interaction in Main

andrey-dru-me1 committed:
add hashCode and equals methods to Cook and Courier classes, so they are equals if they have same name and experience / trunkSize

andrey-dru-me1 committed:
improve main tests

andrey-dru-me1 committed:
add help message

andrey-dru-me1 committed:
rename pizzeria close method to stop, so there is a pair of same-type named methods start-stop;
update uml diagram

andrey-dru-me1 commented:
![image](https://user-images.githubusercontent.com/91488312/229191113-f26f44c6-337c-4598-ae6d-0202b2d0a7ec.png)
